src/main/java/com/redhat/devtools/lsp4ij/features/workspaceSymbol/LSPWorkspaceSymbolSupport.java

```diff
@@ -54,16 +54,26 @@ protected CompletableFuture<List<WorkspaceSymbolData>> doLoad(WorkspaceSymbolPar
                                                                                             @NotNull WorkspaceSymbolParams params,
                                                                                             @NotNull CancellationSupport cancellationSupport) {
         return getLanguageServers(project,
-                        f -> f.getWorkspaceSymbolFeature().isEnabled(),
-                        f -> f.getWorkspaceSymbolFeature().isSupported())
+                f -> f.getWorkspaceSymbolFeature().isEnabled(),
+                f -> f.getWorkspaceSymbolFeature().isSupported())
                 .thenComposeAsync(languageServers -> {
+                    // Remove language servers that aren't supported efficiently by the provided params

```

Comment:
And this is where we use the discriminator to filter out any language servers that don't support types-only queries efficiently when invoked from the Go To Class action. I'm assuming the provided collection should be considered immutable so I'm making a mutable copy before removing filtered language servers.